diff -ruN host-mtd-1.5.1/mkfs.ubifs/hashtable/hashtable_itr.c host-mtd-1.5.1_new/mkfs.ubifs/hashtable/hashtable_itr.c
--- host-mtd-1.5.1/mkfs.ubifs/hashtable/hashtable_itr.c	2014-04-07 13:01:36.000000000 +0200
+++ host-mtd-1.5.1_new/mkfs.ubifs/hashtable/hashtable_itr.c	2018-09-12 03:33:45.748440308 +0200
@@ -38,13 +38,13 @@
 /* key      - return the key of the (key,value) pair at the current position */
 /* value    - return the value of the (key,value) pair at the current position */
 
-void *
+/*void *
 hashtable_iterator_key(struct hashtable_itr *i)
 { return i->e->k; }
 
 void *
 hashtable_iterator_value(struct hashtable_itr *i)
-{ return i->e->v; }
+{ return i->e->v; }*/
 
 /*****************************************************************************/
 /* advance - advance the iterator to the next element
diff -ruN host-mtd-1.5.1/mkfs.ubifs/hashtable/hashtable_itr.h host-mtd-1.5.1_new/mkfs.ubifs/hashtable/hashtable_itr.h
--- host-mtd-1.5.1/mkfs.ubifs/hashtable/hashtable_itr.h	2014-04-07 13:01:36.000000000 +0200
+++ host-mtd-1.5.1_new/mkfs.ubifs/hashtable/hashtable_itr.h	2018-09-12 03:33:44.248425330 +0200
@@ -28,7 +28,7 @@
 /* hashtable_iterator_key
  * - return the value of the (key,value) pair at the current position */
 
-extern inline void *
+static inline void *
 hashtable_iterator_key(struct hashtable_itr *i)
 {
     return i->e->k;
@@ -37,7 +37,7 @@
 /*****************************************************************************/
 /* value - return the value of the (key,value) pair at the current position */
 
-extern inline void *
+static inline void *
 hashtable_iterator_value(struct hashtable_itr *i)
 {
     return i->e->v;
